http/typescript/fetch/package.json

```diff
@@ -2,51 +2,84 @@
     "name": "@microsoft/kiota-http-fetchlibrary",
     "version": "1.0.17",
     "description": "Kiota request adapter implementation with fetch",
-    "main": "dist/cjs/index.js",
-    "module": "dist/es/index.js",
-    "types": "dist/cjs/index.d.ts",
-    "scripts": {
-        "build": "npm run build:cjs && npm run build:esm",
-        "build:cjs": "tsc -p tsconfig.cjs.json",
-        "build:esm": "tsc -p tsconfig.es.json",
-        "test": "echo \"Error: no test specified\" && exit 1",
-        "lint": "eslint . --ext .ts",
-        "lint:fix": "eslint . --ext .ts --fix",
-        "clean": "rm -r ./dist"
-    },
-    "repository": {
-        "type": "git",
-        "url": "git://github.com/microsoft/kiota.git"
-    },
     "keywords": [
         "Kiota",
         "OpenAPI",
         "HTTP",
         "fetch"
     ],
-    "author": "Microsoft",
-    "license": "MIT",
+    "homepage": "https://github.com/microsoft/kiota#readme",
     "bugs": {
         "url": "https://github.com/microsoft/kiota/issues"
     },
-    "homepage": "https://github.com/microsoft/kiota#readme",
-    "publishConfig": {
-        "registry": "https://npm.pkg.github.com"
+    "repository": {
+        "type": "git",
+        "url": "git://github.com/microsoft/kiota.git"
+    },
+    "license": "MIT",
+    "author": "Microsoft",
+    "main": "dist/cjs/src/index.js",
+    "browser": {
+        "./dist/es/src/index.js": "./dist/es/src/browser/index.js",
+        "./dist/es/src/utils/utils.js": "./dist/es/src/utils/browser/utils.js",
+        "./dist/es/src/middlewares/defaultFetchHandler.js": "./dist/es/src/middlewares/browser/defaultFetchHandler.js",
+        "./dist/es/src/middlewares/middlewareFactory.js": "./dist/es/src/middlewares/browser/middlewareFactory.js"
+    },
+    "types": "dist/cjs/src/index.d.ts",
+    "files": [
+        "dist/",
+        "src/"
+    ],
+    "scripts": {
+        "build": "npm run build:cjs && npm run build:es",
+        "build:cjs": "tsc -p tsconfig.cjs.json",
+        "build:es": "tsc -p tsconfig.es.json",
+        "clean": "rm -r ./dist",
+        "karma": "npm run rollup && karma  start --single-run --browsers ChromeHeadless karma.conf.js",
+        "lint": "eslint . --ext .ts",
+        "lint:fix": "eslint . --ext .ts --fix",
+        "rollup": "rollup -c",
+        "test": "npm run test:cjs && npm run test:es && npm run karma",
+        "test:cjs": "npm run build:cjs   && mocha  'dist/cjs/test/common/**/*.js' && mocha  'dist/cjs/test/node/**/*.js'",
+        "test:es": " npm run build:es && mocha 'dist/es/test/common/**/*.js'  --require  esm && mocha 'dist/es/test/node/**/*.js'  --require  esm"
     },
     "dependencies": {
-        "@microsoft/kiota-abstractions": "^1.0.30",
-        "cross-fetch": "^3.1.5",
-        "tslib": "^2.3.1"
+        "@microsoft/kiota-abstractions": "1.0.31",
+        "node-fetch": "^2.6.5",
```

Comment:
any specific reason we're not on 3.X anymore? I believe the package size issues have been addressed since our last discussions. https://www.npmjs.com/package/node-fetch

Reply:
3.X is ESM-only.